app/helpers/application_html_formatters_helper.rb

```diff
@@ -74,7 +75,9 @@ def format_code_block(code, language = nil)
       end
     end
 
-    format_with_pipeline(DefaultCodePipeline, code)
+    formatted_code = DefaultCodePipeline.to_document(code)
```

Comment:
I get what you mean, but the pipeline still need to be created with the `code` as an argument and executed in two steps ( we cannot get a instance of pipeline which supports line start numbers and simply call it ):
```ruby
def format_code_with_line_numbers(code, line_number)
   formatted_code = DefaultCodePipeline.to_document(code)
   PreformattedTextLineNumbersFilter.new(formatted_code, line_start).call
end
```
What I can think of is to move `code_with_line_numbers` as separate method, and deals with the pipeline details there.